src/queue.c

```diff
@@ -77,12 +82,21 @@ static int _dispatch_pthread_sigmask(int how, sigset_t *set, sigset_t *oset);
 
 #if DISPATCH_COCOA_COMPAT
 static dispatch_once_t _dispatch_main_q_port_pred;
-static dispatch_queue_t _dispatch_main_queue_wakeup(void);
 unsigned long _dispatch_runloop_queue_wakeup(dispatch_queue_t dq);
 static void _dispatch_runloop_queue_port_init(void *ctxt);
 static void _dispatch_runloop_queue_port_dispose(dispatch_queue_t dq);
 #endif
 
+#if DISPATCH_COCOA_COMPAT || DISPATCH_LINUX_COMPAT
```

Comment:
`DISPATCH_LINUX_COMPAT` is not the right name for this, it is about providing runloop backends, you should introduce a `DISPATCH_ENABLE_RUNLOOP_SUPPORT` and make sure that `DISPATCH_COCOA_COMPAT` implies it.

then fix all places that need it to use that new guard.